pdf.go

```diff
@@ -859,6 +859,45 @@ func (w *pdfPageWriter) EndTextObject() {
 	w.inTextObject = false
 }
 
+/* Extract from the pdf specification:
+When a string is written by enclosing the data in parentheses, bytes whose values are the
+same as thoseof  the  ASCII  characters  left  parenthesis  (40),  right  parenthesis  (41),
+and  back-slash  (92)  must  be  preceded  by  a  backslash  character. (...)
+These  rules  apply  to  each individual  byte  in  a  string  object, whether  the  string
+is  interpreted  by  the  text-showing operators as single-byte or multiple-byte character codes.
+*/
+type stringBytesEscaper struct {
+	w           io.Writer
+	escapeChars []byte
+}
+
+func (r stringBytesEscaper) Write(p []byte) (n int, err error) {
+	var k int
+	i := 0
+	for j, b := range p {
+		if b != '(' && b != ')' && b != '\\' {
+			continue
+		}
+		if i != j {
```

Comment:
This is the same as `i != 0` which might be easier to reason about

Reply:
You meant `j != 0` (otherwise the tests fail :-)

Reply:
True, you're right!